src/node_sockaddr.cc

```diff
@@ -215,7 +215,10 @@ bool in_network_ipv4(
     const SocketAddress& ip,
     const SocketAddress& net,
     int prefix) {
-  uint32_t mask = ((1 << prefix) - 1) << (32 - prefix);
+  if (prefix == 32)
+    return compare_ipv4(ip, net) == SocketAddress::CompareResult::SAME;
+
+  uint32_t mask = ((1ull << prefix) - 1) << (32 - prefix);
```

Comment:
```suggestion
  uint32_t mask = ((1ull << prefix) - 1) << (32 - prefix);
```
The check isn't needed when the mask is computed correctly.